vpr/src/util/vpr_utils.cpp

```diff
@@ -633,33 +631,51 @@ bool is_io_type(t_physical_tile_type_ptr type) {
 bool is_empty_type(t_physical_tile_type_ptr type) {
     auto& device_ctx = g_vpr_ctx.device();
 
-    return type == device_ctx.EMPTY_TYPE;
+    return type == device_ctx.EMPTY_PHYSICAL_TILE_TYPE;
+}
+
+bool is_empty_type(t_logical_block_type_ptr type) {
+    auto& device_ctx = g_vpr_ctx.device();
+
+    return type == device_ctx.EMPTY_LOGICAL_BLOCK_TYPE;
 }
 
 t_physical_tile_type_ptr physical_tile_type(t_logical_block_type_ptr logical_block_type) {
```

Comment:
I'm not sure this function ever makes sense.  Can you explain how it is valid?

Reply:
So, with the clean-up of the equivalent-sites PR I have been trying to get rid of these functions:
- `physical_tile_type(t_logical_block_type_ptr logical_block);`
- `logical_block_type(t_physical_tile_type_ptr physical_tile);`

And leave only the one that is needed when accessing the physical tile given a clustered block ID, after placement.

There still are some issues to be solved in order to get rid of these functions (mainly because some parts of the code are still "thinking" that there is no distinction between physical and logical types).

Anyway, I am working at a solution that does not require these functions anymore.